tool/persistentqueue/diskqueue/diskqueue.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package diskqueue
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/nsqio/go-diskqueue"
+
+	"github.com/aws/amazon-cloudwatch-agent/tool"
+	"github.com/aws/amazon-cloudwatch-agent/tool/persistentqueue"
+)
+
+type diskQueue struct {
+	marshal   persistentqueue.Marshaler
+	unmarshal persistentqueue.Unmarshaler
+	queue     diskqueue.Interface
+	size      int64
+	logger    telegraf.Logger
```

Comment:
nit - I think we should transition from the telegraf logger to zap

Reply:
sure, I will change it right now

Reply:
but this logger parameter is come from `func NewPusher(target Target, service CloudWatchLogsService, flushTimeout time.Duration, retryDuration time.Duration, logger telegraf.Logger, stop <-chan struct{}, wg *sync.WaitGroup, agentInfo agentinfo.AgentInfo)`, and the `telegraf.Logger ` cannot convert to `zap.Logger`,
so i have to use `telegraf.Logger`, right ? or I can just not pass this `telegraf.Logger` parameter to diskQueue and directly create a logger using zap.Logger?

how do you think

Reply:
hi, andrew, how do you think , do I need to directly create a new `zap.Logger` in `diskQueue` without pass it as parameter from `NewPusher`

Reply:
Ok if there's no zap logger to pass in, using the telegraf one is fine I guess